data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -276,6 +276,12 @@ public String toJsonString() {
         return jsonNode.toString();
     }
 
+    @Override
+    public String toJsonStringWithTags(final String tagsKey) {
+        String result = jsonNode.toString();
+        return result.substring(0, result.length()-1) + ",\""+tagsKey+"\":"+ getMetadata().getTags()+"}";
```

Comment:
Does this need quotes around the `getMetadata().getTags`? Right now the final json would look like this

```
end_of_event, "tagsKey": [tag1, tag2]}
```

Should it look like this instead? I'm not sure if it's a requirement from json but looking at this (https://www.microfocus.com/documentation/silk-performer/205/en/silkperformer-205-webhelp-en/GUID-0847DE13-2A2F-44F2-A6E7-214CD703BF84.html) seems like we want the following

```
end_of_event, "tagsKey": ["tag1", "tag2"]}
```

Reply:
We could do that. By default `toString()` of a `Set` does not put quotes around the individual tags. If we want quotes around each tag, then we will have to manually generate it like that. I am OK, if that's what we want.

Reply:
Can this JsonString be re-serialized into an event in its current form?

Reply:
I don't think so. It's one way (ie event to string with tags) when we want to send to sink with tags.